### ISSUE TITLE
Support Orleans 3.3.0

### ISSUE THREAD

**Issue by erdtsieck** [OrleansContrib/Orleans.EventSourcing.Snapshot#3]
Currently running the silohost throws an exception while using Orleans 3.3.0

```
fail: Orleans.LifecycleSubject[100450]
      Lifecycle start canceled due to errors at stage 1000: System.TypeLoadException: Method 'get_RecordExists' in type 'Orleans.EventSourcing.Snapshot.SnapshotStateWithMetaDataAndETag`2' from assembly 'Orleans.EventSourcing.Snapshot, Version=1.0.3.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
         at Orleans.EventSourcing.Snapshot.LogViewAdaptor`2.InitializeConfirmedView(TLogView initialstate)
         at Orleans.EventSourcing.Common.PrimaryBasedLogViewAdaptor`3..ctor(ILogViewAdaptorHost`2 host, TLogView initialstate, ILogConsistencyProtocolServices services)
         at Orleans.EventSourcing.Snapshot.LogViewAdaptor`2..ctor(ILogViewAdaptorHost`2 host, TLogView initialState, IGrainStorage grainStorage, String grainTypeName, Func`2 snapshotStrategy, ILogConsistencyProtocolServices services, Boolean useIndependentEventStorage, IGrainEventStorage eventStorage)
         at Orleans.EventSourcing.Snapshot.LogConsistencyProvider.MakeLogViewAdaptor[TLogView,TLogEntry](ILogViewAdaptorHost`2 hostGrain, TLogView initialState, String grainTypeName, IGrainStorage grainStorage, ILogConsistencyProtocolServices services)
         at Orleans.EventSourcing.JournaledGrain`2.InstallAdaptor(ILogViewAdaptorFactory factory, Object initialState, String graintypename, IGrainStorage grainStorage, ILogConsistencyProtocolServices services)
         at Orleans.LogConsistency.LogConsistentGrain`1.InstallLogViewAdaptor(Factory`2 protocolServicesFactory, ILogViewAdaptorFactory factory, IGrainStorage grainStorage)
         at Orleans.LogConsistency.LogConsistentGrain`1.OnSetupState(CancellationToken ct)
         at Orleans.LifecycleExtensions.Observer.OnStart(CancellationToken ct)
         at Orleans.LifecycleSubject.<OnStart>g__CallOnStart|7_0(OrderedObserver observer, CancellationToken cancellationToken)
```

**Comment by zh6335901** [comment]
Thanks for you contribution! I have some comment to your pull request, Can you look it?

**Comment by zh6335901** [comment]
@annemartijn0 Merged, And published it to version 1.0.4, thanks!
